check/src/typecheck.rs

```diff
@@ -1235,8 +1237,8 @@ impl<'a> Typecheck<'a> {
         debug!("Unify {} <=> {}",
                types::display_type(&self.symbols, expected),
                types::display_type(&self.symbols, &actual));
-        let mut state = ::unify_type::State::new(&self.environment);
-        match unify::unify(&self.subs, &mut state, expected, &actual) {
+        let state = ::unify_type::State::new(&self.environment);
```

Comment:
There were a bunch of other places with `::unify_type::*` so I added `use unify_type;` at the top instead. `unify_type` is not an ideal name but just using `unify` may confuse it with the `unify` module itself.